dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/WorkflowsAdmin.java

```diff
@@ -189,4 +213,92 @@ public Response deleteIpWhitelist() {
         return ok("Restored whitelist to default (127.0.0.1;::1)");
     }
 
+    @POST
+    @Path("rerun")
+    public Response rerunWorkflow(@QueryParam("type") String type) {
+        try {
+            AuthenticatedUser user = findAuthenticatedUserOrDie();
+            if (!user.isSuperuser()) {
+                return error(Response.Status.FORBIDDEN, "Only superusers can run workflow");
+            }
+            if (type == null) {
+                return error(Response.Status.BAD_REQUEST, "Missing processing type - 'type=failedOnly' or 'type=notPerformedOnly' is required");
+            }
+
+            int datasetProcessed = 0;
+            for (Dataset dataset:datasetRepository.findAll()) {
+                if (dataset.isLockedFor(DatasetLock.Reason.Workflow)) {
```

Comment:
Imo - check should be little later. Now we will return error if dataset have running workflow.